src/interact.js

```diff
@@ -881,21 +886,21 @@ class Interact {
      * Broadcasts the `TargetSelectionStarts` and the `TargetSelected` event. Can broadcast the `CastSpellOnMinion` event.
      * 
      * @param {string} prompt The prompt to ask
-     * @param {boolean | string} [elusive=false] Whether or not to prevent selecting elusive minions, if this is a string, allow selecting elusive minions but don't trigger secrets / quests
+     * @param {Card | null} card The card that called this function.
      * @param {"enemy" | "friendly" | null} [force_side=null] Force the user to only be able to select minions / the hero of a specific side: ["enemy", "friendly"]
      * @param {"hero" | "minion" | null} [force_class=null] Force the user to only be able to select a minion or a hero: ["hero", "minion"]
-     * @param {string[]} [flags=[]] Change small behaviours ["allow_locations" => Allow selecting location, ]
+     * @param {import('./types').SelectTargetFlags[]} [flags=[]] Change small behaviours ["allow_locations" => Allow selecting location, ]
      * 
      * @returns {Card | Player | false} The card or hero chosen
      */
-    selectTarget(prompt, elusive = false, force_side = null, force_class = null, flags = []) {
+    selectTarget(prompt, card, force_side = null, force_class = null, flags = []) {
         // force_class = [null, "hero", "minion"]
         // force_side = [null, "enemy", "friendly"]
 
-        game.events.broadcast("TargetSelectionStarts", [prompt, elusive, force_side, force_class, flags], game.player);
-        let target = this._selectTarget(prompt, elusive, force_side, force_class, flags);
+        game.events.broadcast("TargetSelectionStarts", [prompt, card, force_side, force_class, flags], game.player);
+        let target = this._selectTarget(prompt, card, force_side, force_class, flags);
 
-        game.events.broadcast("TargetSelected", target, game.player);
+        if (target) game.events.broadcast("TargetSelected", [card, target], game.player);
```

Comment:
I have the `if (target)` here since according to the `events.txt` file, the target should always be of type `Card | Player`